app/Http/Controllers/SubmissionController.php

```diff
@@ -151,29 +152,30 @@ private function submitProcess(): Response
         } catch (BadSubmissionException $e) {
             $xml_info['xml_handler'] = '';
             $message = "Could not determine submission file type for: '{$stored_filename}'";
-            Log::warning($message);
-            if ((bool) config('cdash.validate_xml_submissions') === true) {
-                $this->failProcessing($filename, 400, $message);
-            }
+            Log::info($message);
+            $this->failProcessing($filename, 400, $message);
         }
+
         if ($xml_info['xml_handler'] !== '') {
             // If validation is enabled and if this file has a corresponding schema, validate it
             $validation_errors = [];
             try {
                 $validation_errors = $xml_info['xml_handler']::validate($stored_filename);
             } catch (FileNotFoundException|UnableToReadFile $e) {
-                Log::warning($e->getMessage());
-                if ((bool) config('cdash.validate_xml_submissions') === true) {
-                    $this->failProcessing($filename, 400, "XML validation failed for $filename:" . PHP_EOL . $e->getMessage());
-                }
+                $message = $e->getMessage();
+                Log::info($message);
+                $this->failProcessing($filename, 500, "Unable to read file for validation $filename:" . PHP_EOL . $e->getMessage());
```

Comment:
Optional: you could consider just not catching these errors at all, letting the exceptions fall through and end up as stack traces in the log.

Reply:
I wonder if we're trying to add in the filename or some other piece of information that we didn't have before.  I've removed the check and I don't notice much of a difference.  Thanks for the suggestion!

Reply:
Good point.  We should definitely make sure the filename is included in any error messages coming from this location.

Reply:
It looks like it does: https://github.com/Kitware/CDash/blob/bd286614b2a2bd63dd77fc3f64ce8004e06bafe0/app/Utils/SubmissionUtils.php#L98